geode-assembly/src/acceptanceTest/java/org/apache/geode/launchers/LocatorLauncherWithPulseAcceptanceTest.java

```diff
@@ -0,0 +1,113 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.launchers;
+
+import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.junit.rules.RequiresGeodeHome;
+
+public class LocatorLauncherWithPulseAcceptanceTest {
+
+  private static final String LOCATOR_NAME = "the-locator";
+
+  private int locatorPort;
+  private int httpServicePort;
+  private Process locator;
+  private Path geodeDependencies;
+  private Path stdoutFile;
+  private Path locatorLogFile;
+  private Path pulseLogFile;
+
+  @Rule
+  public RequiresGeodeHome requiresGeodeHome = new RequiresGeodeHome();
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void setUpGeodeDependencies() {
+    Path geodeHome = requiresGeodeHome.getGeodeHome().toPath();
+    geodeDependencies = geodeHome.resolve("lib/geode-dependencies.jar");
+
+    assertThat(geodeDependencies).exists();
+  }
+
+  @Before
+  public void setUpOutputFiles() {
+    stdoutFile = temporaryFolder.getRoot().toPath().resolve("stdout.txt");
+    locatorLogFile = temporaryFolder.getRoot().toPath().resolve(LOCATOR_NAME + ".log");
+    pulseLogFile = temporaryFolder.getRoot().toPath().resolve("pulse.log");
+  }
+
+  @Before
+  public void setUpRandomPorts() {
+    int[] ports = AvailablePortHelper.getRandomAvailableTCPPorts(2);
+    locatorPort = ports[0];
+    httpServicePort = ports[1];
+  }
+
+  @After
+  public void stopLocator() throws Exception {
+    if (locator != null) {
+      locator.destroyForcibly().waitFor(4, TimeUnit.SECONDS);
+    }
+  }
+
+  @Test
+  public void locatorLauncherStartsPulse() throws Exception {
+    ProcessBuilder processBuilder = new ProcessBuilder()
+        .redirectErrorStream(true)
+        .redirectOutput(stdoutFile.toFile())
+        .directory(temporaryFolder.getRoot())
+        .command("java",
+            "-Dgemfire.http-service-port=" + httpServicePort,
+            "-Dgemfire.jmx-manager-start=true",
+            "-Djava.awt.headless=true",
+            "-cp", geodeDependencies.toFile().getAbsolutePath(),
+            "org.apache.geode.distributed.LocatorLauncher", "start", LOCATOR_NAME,
+            "--port", String.valueOf(locatorPort));
+
+    System.out.println("Launching command: " + processBuilder.command());
+
+    locator = processBuilder
+        .start();
+
+    assertThat(locator.isAlive()).isTrue();
+
+    await().untilAsserted(() -> {
+      LogFileAssert.assertThat(locatorLogFile.toFile())
+          .exists()
```

Comment:
Why `locatorLogFile` exits for this test, while ``locatorLogFile` doesn't exist for the other tests in this pull request? Can you share what configuration makes this difference?

Reply:
LocatorLauncherWithPulseAcceptanceTest is the only test in this PR that doesn't provide a custom log4j2 xml file. So that means, it'll create a locator log file.

The others tests all use a custom log4j2 xml file which turns off logging to a file and turns on logging to stdout. These tests assert that the locator log file doesn't exist to help verify that the custom log4j2 xml file for that test was actually used.